### PR TITLE
[datafusion-1305]Extract EmptyRelation, Limit, Values in LogicalPlan

### LINKED ISSUE #1305: Extract `EmptyRelation`, `Limit`, `Values` in LogicalPlan as independent struct
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
There are some discussions in #1228

**Describe the solution you'd like**
like #1290

**Describe alternatives you've considered**
The task is mechanical, however, it is ideal for newcomers to familiarize themselves with datafusion's build, test, and PR processes.

**Additional context**
If you want to do it, please ping me and assign the task to you, avoiding collisions with other people.

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

Closes #1305 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by liukun4515** [comment]
@alamb  PTAL

File: datafusion/src/logical_plan/plan.rs [hunk 1]
```diff
@@ -263,19 +292,9 @@ pub enum LogicalPlan {
...
-    },
+    EmptyRelation(EmptyRelation),
     /// Produces the first `n` tuples from its input and discards the rest.
-    Limit {
-        /// The limit
-        n: usize,
-        /// The logical plan
-        input: Arc<LogicalPlan>,
-    },
+    Limit(Limit),
```
Comment (xudong963) [inline datafusion/src/logical_plan/plan.rs:297]: Using a different name will be better, such as `LimitPlan`, keep the same as others.

File: datafusion/src/optimizer/constant_folding.rs [hunk 2]
```diff
@@ -107,7 +107,7 @@ impl OptimizerRule for ConstantFolding {
 
                 utils::from_plan(plan, &expr, &new_inputs)
             }
-            LogicalPlan::TableScan { .. } | LogicalPlan::EmptyRelation { .. } => {
+            LogicalPlan::TableScan { .. } | LogicalPlan::EmptyRelation(_) => {
```
Comment (xudong963) [inline datafusion/src/optimizer/constant_folding.rs:110]: nit: keep consistent.   `{ .. }` `(_)`

**Review by xudong963 — COMMENTED** [review]
Thanks @liukun4515

File: datafusion/src/logical_plan/plan.rs (hunk 1 above)
Comment (alamb) [reply datafusion/src/logical_plan/plan.rs:297]: I actually think the way this PR is written, `Limit(Limit)` is more consistent.  The only one that seems to follow a  different pattern is `TableScanPlan`  (which we should perhaps rename to `TableScan`?)

**Review by alamb — APPROVED** [review]
I think it looks good. Thanks @liukun4515 !

File: datafusion/src/optimizer/constant_folding.rs (hunk 2 above)
Comment (alamb) [reply datafusion/src/optimizer/constant_folding.rs:110]: perhaps we can do this in a follow on pr. Good spot @xudong963

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Merged this one in to minimize conflicts with other outstanding PRs

File: datafusion/src/optimizer/constant_folding.rs (hunk 2 above)
Comment (liukun4515) [reply datafusion/src/optimizer/constant_folding.rs:110]: @xudong963 @alamb 
I might miss this point when write the code.
After all the extracting logical plan, i will file a followup pull request to keep consistent.

Comment (alamb) [reply datafusion/src/optimizer/constant_folding.rs:110]: (I merged the PR in before you had a chance to respond to this comment -- so it was my fault :) )

**Pull request merged**